cephfs: adjust RBAC for CephFS provisioner role

humblec opened the pull request:
CephFS CSI driver does not need permissions on Node,ConfigMap objects. 


Updates: #3018

nixpanic commented on deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml:
```diff
@@ -30,25 +27,25 @@ rules:
   - apiGroups: ["snapshot.storage.k8s.io"]
     resources: ["volumesnapshots"]
-    verbs: ["get", "list"]
+    verbs: ["get", "list", "watch"]
```
why is this added? explain it in the commit message please

humblec replied:
not needed.. removed.

nixpanic commented on deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml:
```diff
@@ -30,25 +27,25 @@ rules:
   - apiGroups: [""]
     resources: ["persistentvolumeclaims/status"]
-    verbs: ["update", "patch"]
+    verbs: ["patch"]
```
does this require a certain version of other Kubernetes components?

humblec replied:
all supported versions of kube should work here.

nixpanic commented on deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml:
```diff
@@ -78,9 +75,6 @@ metadata:
   name: cephfs-external-provisioner-cfg
 rules:
   # remove this once we stop supporting v1.0.0
```
does this comment not refer to the ConfigMaps section? if that is the case, remove it as well

humblec replied:
yep, removed that as well.

nixpanic reviewed (dismissed):
instead of the comment that these are 'unwanted', a more detailed explanation would be much appreciated in the commit message. Were these never needed, or have components changed to use a new API (like update -> patch), which should mention the version of the component where the change was done.

humblec commented:
@Madhu-1 ptal.. thanks..

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.21

humblec commented:
@Mergifyio refresh

humblec commented:
@Mergifyio rebase

Madhu-1 commented on deploy/cephfs/kubernetes/csi-provisioner-rbac.yaml:
```diff
@@ -36,19 +33,19 @@ rules:
   - apiGroups: ["storage.k8s.io"]
     resources: ["volumeattachments"]
-    verbs: ["get", "list", "watch", "update", "patch"]
+    verbs: ["get", "list", "watch", "patch"]
```
as the discussion is going on in the kubernetes-csi repo. keeping this on hold as well. https://github.com/kubernetes-csi/external-attacher/pull/354

humblec replied:
@Madhu-1 I have removed the update/patch for now. please revisit

humblec committed:
cephfs: adjust RBAC for CephFS provisioner role

CephFS CSI driver does not need permissions on Node,ConfigMap objects.
This PR also adjust the update->Patch for a couple of RBAC.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@Madhu-1 can you bring back the the approval?
